### DISCUSSION TITLE
Uncaught TypeError

### DISCUSSION THREAD

**Discussion by Ali-Zmn** [data-driven-forms/react-forms#1232] [Need help] [answered]
Sometimes when my initial values in `FormRenderer` does not include all fields, I face this error:
`Uncaught TypeError: Cannot read properties of undefined (reading 'initial')
    at eval (condition.js?3893:72:1)`
I would assume if the initial value is not in `FormRenderer initialValues` it would pick the `Schema initialValue` and if no initial value is provided in the schema, it should keep it empty, right?

I couldn't reproduce it in codesandbox and in my code, if I rerender my component (reload) a couple of times, it works fine!
Do you have any idea?

![Screen Shot 2022-04-12 at 11 24 39](https://user-images.githubusercontent.com/87710154/162997643-d99514d9-73cf-49d0-a3ef-32650f95ddf8.png)

**Comment by rvsia** [comment]
Seems that it happens here: https://github.com/data-driven-forms/react-forms/blob/bb71375b7115cf952d15b8608bc6e487b3ac85d2/packages/react-form-renderer/src/condition/condition.js#L57

I guess there could be some problem when a condition is triggered during a render or so. We will take a look tomorrow.

**Reply by Ali-Zmn** [reply]
I also have this issue that when a condition is satisfied, it still prefers the schema initial value rather than the condition setter!

[in this code](https://codesandbox.io/s/react-data-driven-initialvalue-sd2o07?file=/src/components/schema.js:711-724), I reproduce the `Cannot read properties of undefined (reading 'initial')` (by choosing anything except B in field 1) and also the setter behaviour, which should replace `schema initial value` with `set with then`

Currently, I overcome the setter issue by removing the initial value from the schema and adding `else` statement, but I think it's just a temporary workaround

**Reply by rvsia** [reply]
The first issue is caused by `clearOnUnmount` and by that the field state is removed from the state before the condition is calculated - we will fix this.

However, the second issue is a feature by design: on the initial render, setters with initial values are ignored, I thought this is something you wanted: https://github.com/data-driven-forms/react-forms/discussions/1223

**Reply by rvsia** [reply]
Maybe we could add some flag to the condition to set behavior on initialValue, what do you think @Hyperkid123 ?

**Reply by Hyperkid123** [reply]
@rvsia  I am definitely against changing the default behavior. That has the potential to mess up many existing forms.

If necessary we can add a flag as you suggested with the standard priority. Schema prop > renderer props.

**Reply by rvsia** [reply]
On other hand, consumers can just remove the initial value from their schemas/initial values manually with no need to add some flag.

**Reply by Hyperkid123** [reply]
Yeah, that is why said if necessary.

**Reply by Ali-Zmn** [reply]
> On other hand, consumers can just remove the initial value from their schemas/initial values manually with no need to add some flag.

Make sense! If it is the default behaviour, I'm with you.
Thanks

**Comment by rvsia** [comment, marked as answer]
The uncaught error fixed here https://github.com/data-driven-forms/react-forms/pull/1234
